src/dao/user.dao.ts

```diff
@@ -28,6 +28,31 @@ class UserDAO {
     });
   }
 
+  async fetchUsers(
+    filter: { username?: string; email?: string },
+    page = 1,
+    pageSize = 10
+  ) {
+    const skip = (page - 1) * pageSize;
+    const hasFilters = Object.values(filter).some((value) => value);
+
+    const whereCondition = [{
+      email: { contains: filter.email },
+    }, {
+      username: { contains: filter.username },
+    },];
+
+    const where: Prisma.UserWhereInput = {
+      OR: hasFilters ? whereCondition : undefined,
+    };
+
```

Comment:
Some things I don't love in prisma. `Undefined` is considered a no filter parameter. Meaning, if you pass undefined to a where condition it will ignore that column in the where. This works, but an `OR: []` or an `OR: [undefined condition, undefined condition]`  does not generate an empty where, it instead generates a where 1=0, so no data is returned.

So I cannot simply set the where and pass undefined as the search attributes, we need to check if any of them is true  and if not, set undefined directly to the `or` codition

Reply:
Other things, to add a count to a find many best way is to create a client extension 😞 No upsert out of the box 😢

Reply:
https://orm.drizzle.team/docs/overview My future ~ORM~ Query Builder to test

Reply:
Nice I will check it